lib/srv/forward/sshserver.go

```diff
@@ -999,7 +999,13 @@ func (s *Server) handleSessionChannel(ctx context.Context, nch ssh.NewChannel) {
 				),
 			)
 
-			if err := s.dispatch(ctx, ch, req, scx); err != nil {
+			// some functions called inside dispatch() may handle replies to SSH channel requests internally,
+			// rather than leaving the reply to be handled inside this loop. in that case, those functions are
+			// expected to return a ReplyAlreadyHandled error to halt the default behaviour.
+			err := s.dispatch(ctx, ch, req, scx)
+			if trace.IsReplyAlreadyHandled(err) {
+				req.WantReply = false
```

Comment:
Instead of modifying the `trace`, returning the error, and setting `WantReply` here, can we set `WantReply` inside the `dispatch` and check it here (plus a comment on why this flow is different)? For me, that sounds like the simplest implementation. 
If you want to relay on an error, I'd just define a new error here instead of modifying `trace`.